assessmentModel/src/commonMain/kotlin/recorders/MotionRecorderConfiguration.kt

```diff
@@ -0,0 +1,142 @@
+        }
+        override fun serialize(encoder: Encoder, value: MotionRecorderType) {
+            encoder.encodeString(value.name.decapitalize())
+        }
+
+        val raw = setOf(
+            Accelerometer,
+            Gyro,
+            Magnetometer
+        )
+        val calculated = setOf(
+            Attitude,
+            Gravity,
+            MagneticField,
+            RotationRate,
+            UserAcceleration
+        )
+    }
+}
+
+/**
+ * A [MotionRecord] is a serializable implementation of a [SampleRecord] that can be used to record a sample from one
+ * of the motion sensors or calculated vectors. All properties on the record ard defined as `null` to allow using this
+ * same serialization object to define both path markers that are added to the data stream to mark step transitions
+ * and the results of the sensor.
+ *
+ * If the [sensorType] is `null` then this is a path marker. Other values included here are only applicable to certain
+ * sensors. This data structure allows for defining all records within a single file.
+ *
+ * Vector properties:
+ * - [x]: The `x` component of the vector measurement for this sensor sample.
+ * - [y]: The `y` component of the vector measurement for this sensor sample.
+ * - [z]: The `z` component of the vector measurement for this sensor sample.
+ * - [w]: The `w` component of the vector measurement for this sensor sample. (Used by the attitude quaternion.)
+ *
+ * Additional properties:
+ * - [eventAccuracy]: A number marking the sensor accuracy of the magnetic field sensor.
+ * - [referenceCoordinate]: Used for an [MotionRecorderType.Attitude] record type to describe the reference frame.
+ * - [heading]: The angle in the range [0,360) degrees with respect to [AttitudeReferenceFrame.XMagneticNorthZVertical].
+ *
+ */
+@Serializable
+data class MotionRecord(
+    override val stepPath: String? = null,
+    @SerialName("timestampDate")
+    override val timestampDateString: String? = null,
+    override val timestamp: Double? = null,
+    val uptime: Double? = null,
+    val sensorType: MotionRecorderType? = null,
+    val x: Double? = null,
+    val y: Double? = null,
+    val z: Double? = null,
+    val w: Double? = null,
+    val eventAccuracy: Int? = null,
+    val referenceCoordinate: AttitudeReferenceFrame? = null,
+    val heading: Double? = null
+) : SampleRecord
+
+@Serializable
+enum class AttitudeReferenceFrame  {
+     /**
+      * Describes a reference frame in which the Z axis is vertical and the X axis points in an arbitrary direction in
+      * the horizontal plane.
+      */
+     @SerialName("Z-Up") XArbitraryZVertical,
+
+     /**
+      * Describes a reference frame in which the Z axis is vertical and the X axis points toward magnetic north.
+      *
+      * - note: Using this reference frame may require user interaction to calibrate the magnetometer.
+      */
+     @SerialName("North-West-Up") XMagneticNorthZVertical;
```

Comment:
I expect for most use cases (exception: those involving studies of human sensitivity to Earth's magnetic field) that true north would be more useful, if heading is needed at all. If we're including this, should we include the true north reference frame as an option as well? (would suggest just replacing this one, but it does require location to be available as well as a calibrated magentometer.)

Reply:
Copied directly from the Apple documentation on the sensor.